src/parsers.rs

```diff
@@ -1,22 +1,784 @@
+        InvalidMonthRange => "Month is outside valid range (1-12)".to_string(),
+        InvalidDayRange => "Day is outside valid range for the given month/year".to_string(),
+        DateYear => "Invalid year format".to_string(),
+        DateMonth => "Invalid month format".to_string(),
+        DateDay => "Invalid day format".to_string(),
+        TimeHour => "Invalid hour format".to_string(),
+        TimeMinuteSecond => "Invalid minute or second format".to_string(),
+        TimeSecond => "Invalid second format".to_string(),
+        FractionPart => "Invalid fractional seconds format".to_string(),
+        ParseFloat => "Invalid fractional seconds value".to_string(),
+        AbruptEnd { location } => format!("Unexpected end while parsing {location}"),
+        InvalidEnd => "Unexpected character at end of input".to_string(),
+        _ => format!("Parse error: {err:?}"),
+    };
+    TemporalValidationError::ParseError(message)
+}
+
+// ECMAScript Temporal specific validation
+/// Validates a date record for ECMAScript Temporal year limits
+fn validate_date_record_impl(record: DateRecord) -> Result<IsoDate, TemporalValidationError> {
+    // Only validate ECMAScript Temporal year limits
+    if !year_month_within_limits(record.year, record.month) {
+        return Err(TemporalValidationError::InvalidYear(record.year));
+    }
+
+    Ok(IsoDate::new_unchecked(
+        record.year,
+        record.month,
+        record.day,
+    ))
+}
+
+/// Creates an IsoTime from a time record
+fn validate_time_record_impl(record: TimeRecord) -> Result<IsoTime, TemporalValidationError> {
+    // ixdtf validates time components
+    IsoTime::from_time_record(record)
+        .map_err(|_| TemporalValidationError::ParseError("Invalid time components".to_string()))
+}
+
+/// Parser encoding enum that specifies how temporal strings are encoded.
+#[derive(Debug)]
+pub enum ParserEncoding<'a> {
+    Utf8(&'a [u8]),
+    Utf16(&'a [u16]),
+}
+
+/// Public parser that wraps `IxdtfParser` and enforces Temporal parsing requirements.
+#[derive(Debug)]
+pub struct TemporalParser<'a> {
+    encoding: ParserEncoding<'a>,
+}
+
+impl<'a> TemporalParser<'a> {
+    /// Creates a new `TemporalParser` from UTF-8 bytes.
+    #[inline]
+    pub const fn from_utf8(source: &'a [u8]) -> Self {
+        Self {
+            encoding: ParserEncoding::Utf8(source),
+        }
+    }
+
+    /// Creates a new `TemporalParser` from UTF-16 code units.
+    #[inline]
+    pub const fn from_utf16(source: &'a [u16]) -> Self {
+        Self {
+            encoding: ParserEncoding::Utf16(source),
+        }
+    }
+
+    /// Creates a new `TemporalParser` from a string slice by converting to UTF-8 bytes.
+    #[inline]
+    pub fn from_str_as_utf8(source: &'a str) -> Self {
+        Self::from_utf8(source.as_bytes())
+    }
+
+    /// Parses the source into a `PlainDateTime` compatible record.
+    pub fn parse_date_time(&self) -> TemporalResult<ParsedDateTime<'a>> {
```

Comment:
suggestion: ideally we link to exactly which part of the spec this implements: there are a lot of parseable things called DateTime